shared/domains/chelonia/utils.js

```diff
@@ -48,7 +48,7 @@ export const findContractIDByForeignKeyId = (state: Object, keyId: string): ?str
   if (!keyId || !state?._vm?.authorizedKeys?.[keyId]?.foreignKey) return
 
   try {
-    const fkUrl = new URL(state._vm?.authorizedKeys?.[keyId]?.foreignKey)
+    const fkUrl = new URL(state._vm.authorizedKeys[keyId].foreignKey)
```

Comment:
Now we're confident these will be here? Why?

Reply:
Yeah, your suggestion, or, if it works, something like:

```js
const fk = state?._vm?.authorizedKeys?.[keyId]?.foreignKey
if (!keyId || !fk) return

...
```

Reply:
Your suggestion won't work because `keyId` needs to be checked first.

```js
if (!keyId) return
const fk = state?._vm?.authorizedKeys?.[keyId]?.foreignKey
if (!fk) return
```

This works though.